optillm/entropy_spike_beam.py

```diff
@@ -1,103 +1,160 @@
+import ast
+import asyncclick as click
 import torch
-from transformers import PreTrainedModel, PreTrainedTokenizer
-from typing import List, Dict, Tuple
-
-def get_device():
-    if torch.cuda.is_available():
-        return torch.device("cuda")
-    else:
-        return torch.device("cpu")
-
-def entropy_spike_beam(
-    model: PreTrainedModel,
-    tokenizer: PreTrainedTokenizer,
-    messages: List[Dict[str, str]],
+from openai import AsyncOpenAI
+from tqdm.auto import tqdm
+from transformers import AutoTokenizer
+from typing import List, Dict, Optional, Tuple
+
+
+async def generate_one_token(
```

Comment:
This is sick.. async is mainly for making sure we can send several requests to generate tokens, right ?

Reply:
yeah so that we don't wait for one API call to finish before sending another one